crates/biome_service/src/workspace/server.rs

```diff
@@ -221,6 +226,23 @@ impl WorkspaceServer {
             .unwrap_or(DocumentFileSource::from_path(path))
     }
 
+    fn get_js_syntax(&self, path: &Utf8Path) -> Result<AnyJsRoot, WorkspaceError> {
+        let documents = self.documents.pin();
+        let doc = documents.get(path).ok_or_else(WorkspaceError::not_found)?;
+        let file_source = self.file_sources[doc.file_source_index];
+        match file_source {
+            DocumentFileSource::Js(_) => match &doc.syntax {
+                Ok(parse) => Ok(parse.tree()),
+                Err(error) => Err(WorkspaceError::FileTooLarge(error.clone())),
+            },
+            _ => Err(WorkspaceError::source_file_not_supported(
+                file_source,
+                path.to_string(),
+                path.extension().map(str::to_string),
+            )),
+        }
+    }
```

Comment:
A couple of concerns of mine:
- This ties up to one of my first comments, where potentially any language can be parsed and used for the dependency graph. So, having `get_js_syntax` isn't correct. 
- The PR doesn't explain the use of "file too large being an error", so I just assume things. Raising a `WorkspaceError::FileTooLarge` **again** feels wrong because we already do so in the method `get_parse`.

Reply:
Yeah, I looked again, and I actually don't really do anything with these errors at all. Error handling is expected to be done already so I've simplified this a bit.